roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -450,25 +453,62 @@ impl JobDeclarator {
             if let Ok((receiver, sender, _, _)) =
                 Connection::new(stream, HandshakeRole::Responder(responder)).await
             {
-                let setup_message_from_proxy_jd = receiver.recv().await.unwrap();
-                info!(
-                    "Setup connection message from proxy: {:?}",
-                    setup_message_from_proxy_jd
-                );
-
-                let setup_connection_success_to_proxy = SetupConnectionSuccess {
-                    used_version: 2,
-                    // Setup flags for async_mining_allowed
-                    flags: 0b_0000_0000_0000_0000_0000_0000_0000_0001,
+                let address = addr.unwrap();
```

Comment:
These unwraps are safe, and since we are the ones creating the response, I don't think there is a need for additional error handling as this will always succeed. However, I can add an expect block for better readability in case of an error.

Reply:
I'm not a fan of `unwrap` either, but there's many instances where getting rid of them requires changing breaking APIs, which drastically change the nature of the task at hand

this kind of thing should be addressed during the refactor efforts

for now, I think replacing `unwrap` with `expect` where possible feels like a good start